core/src/executor/evaluate/function.rs

```diff
@@ -429,3 +431,37 @@ pub fn format(name: String, expr: Evaluated<'_>, format: Evaluated<'_>) -> Resul
         value => Err(EvaluateError::UnsupportedExprForFormatFunction(value.into()).into()),
     }
 }
+
+pub fn to_date(name: String, expr: Evaluated<'_>, format: Evaluated<'_>) -> Result<Value> {
+    match expr.try_into()? {
+        Value::Str(expr) => {
+            let format = eval_to_str!(name, format);
+            match chrono::NaiveDate::parse_from_str(&expr, &format) {
+                Ok(a) => Ok(Value::Date(a)),
+                Err(error) => {
+                    let error: ChronoFormatError = error.into();
+                    let error: EvaluateError = error.into();
+                    Err(error.into())
+                }
+            }
```

Comment:
```suggestion
            chrono::NaiveDate::parse_from_str(&expr, &format)
                .map(Value::Date)
                .map_err(ChronoFormatError::err_into)
```

It gets so simple. How is it?

Reply:
It looks way simple than before, I think it also increases readability